docs/concepts/realms.md

```diff
@@ -3,47 +3,196 @@ id: realms
 ---
 
 # Realms
+In Gno.land, realms are entities that are addressable and identifiable by a 
+[Gno address](../reference/standard-library/std/address.md). These can be user 
+realms (EOAs), as well as smart contract realms. Realms have several 
+properties:
+- They can own, receive & send [Coins](./standard-library/coin.md) through the
+[Banker](./standard-library/banker.md) module
+- They can be part of a transaction call stack, as a caller or a callee
+- They can be with or without code - EOAs, or smart contracts
 
-A realm refers to a specific instance of a smart contract that can be written
-in [Gno](./gno-language.md). The potentials of realms are endless - you can create virtually any
-application in your mind with built-in composability,
-transparency, and censorship resistance. Here are some ideas of what you can build with realms:
+Realms are represented by a `Realm` type in Gno:
+```go
+type Realm struct {
+    addr    Address // Gno address in the bech32 format
+    pkgPath string  // realm's path on-chain
+}
+```
+The full Realm API can be found under the 
+[reference section](../reference/standard-library/std/realm.md).
+
+## Smart Contract Realms
+Often simply called `realms`, Gno smart contracts contain Gno code and exist
+on-chain at a specific package path. A package path is the defining identifier
+of a realm, while its address is derived from it.
+
+As opposed to [packages](./packages.md), realms are stateful, meaning they keep
+their state between transactions calls. In practice, global variables in the
+realm code are automatically persisted after a transaction has been executed,
+resulting in the fact that developers do not need to bother with the intricacies 
+of state management and persistence, like they do with other languages.
+
+### On-chain paths
+Since Gno.land is built for full transparency and auditability, all on-chain Gno
+code is open-sourced. You can view realm code by simply going to its path in
+your web browser. For example, to take a look at the `gno.land/r/demo/users` realm,
+used for user registration, by visiting
+[`gno.land/r/demo/users`](https://gno.land/r/demo/users/users.gno).
```

Comment:
A possibly better example of this is to use ABCI queries instead of implying this source code "lives in the web browser"

Reply:
[306afca](https://github.com/gnolang/gno/pull/1469/commits/306afca456c2ca22f370acb204ac82044e69788c)